File: furniture-frontend/src/services/productService.js
import api from "./api";

export const getAllProducts = () => {
  return api.get("/products");
};

export const getProductById = (id) => {
  return api.get(`/products/${id}`);
};

// ADMIN: Add product
export const addProduct = (product) => {
  return api.post("/products", product);
};

// ADMIN: Update product
export const updateProduct = (id, product) => {
  return api.put(`/products/${id}`, product);
};

// ADMIN: Delete product
export const deleteProduct = (id) => {
  return api.delete(`/products/${id}`);
};

File: furniture-backend/src/main/java/com/example/furniture_backend/service/ProductService.java
package com.example.furniture_backend.service;

import com.example.furniture_backend.dto.ProductDTO;

import java.util.List;

public interface ProductService {

    ProductDTO addProduct(ProductDTO productDTO);

    List<ProductDTO> getAllProducts();

    ProductDTO getProductById(Long id);

    ProductDTO updateProduct(Long id, ProductDTO productDTO);

    void deleteProduct(Long id);
}
